graphistry/dgl_utils.py

```diff
@@ -295,85 +306,137 @@ def _convert_edgeDF_to_DGL(self, node_column: Optional[str] = None, weight_colum
         res._index_to_entity = {k: v for v, k in res._entity_to_index.items()}
         # this is a sanity check after _remove_edges_not_in_nodes
-        res._check_nodes_lineup_with_edges(node_column)
+        res._check_nodes_lineup_with_edges()
         return res
 
-
     def _featurize_nodes_to_dgl(
         self,
         res,
         X: pd.DataFrame,
         y: pd.DataFrame,
         use_scaler: str = None,
-        #refeaturize: bool = False,
+        feature_engine: FeatureEngine = "auto",
     ):
         logger.info("Running Node Featurization for DGL Graph")
+        logger.debug(f"=*=*=Input shapes are data: {X.shape}, target: {y.shape}")
+
+        X_enc, y_enc, res = res._featurize_or_get_nodes_dataframe_if_X_is_None(
+            X=X,
+            y=y,
+            use_scaler=use_scaler,
+            feature_engine=resolve_feature_engine(feature_engine),
+        )
+
+        logger.debug(
+            f"=*=*=Encoded Node shapes are data: {X_enc.shape}, target: {y_enc.shape}"
 
-        X_enc, y_enc, _ = res._featurize_or_get_nodes_dataframe_if_X_is_None(
-            X=X, y=y, use_scaler=use_scaler, refeaturize=False
         )
 
         ndata = convert_to_torch(X_enc, y_enc)
         # add ndata to the graph
         res.DGL_graph.ndata.update(ndata)
         res._mask_nodes()
-        return res  # have to return despite inplace flag
+        return res
+
 
     def _featurize_edges_to_dgl(
         self,
         res,
         X: pd.DataFrame,
         y: pd.DataFrame,
         use_scaler: str = None,
-        feature_engine: FeatureEngine = "auto"
-        #refeaturize: bool = False,
+        feature_engine: FeatureEngine = "auto",
     ):
         logger.info("Running Edge Featurization for DGL Graph")
 
-        # res = _featurize_nodes(
-        X_enc, y_enc, _ = self._featurize_or_get_edges_dataframe_if_X_is_None(
-            X=X, y=y, use_scaler=use_scaler, feature_engine=resolve_feature_engine(feature_engine)
+        X_enc, y_enc, res = res._featurize_or_get_edges_dataframe_if_X_is_None(
+            X=X,
+            y=y,
+            use_scaler=use_scaler,
+            feature_engine=resolve_feature_engine(feature_engine),
         )
         
+        logger.debug(
+            f"=*=*=Encoded Edge shapes are data: {X_enc.shape}, target: {y_enc.shape}"
+        )
+
         edata = convert_to_torch(X_enc, y_enc)
         # add edata to the graph
         res.DGL_graph.edata.update(edata)
         res._mask_edges()
-        return res  # have to return despite inplace flag
+        return res
 
-    def build_dgl_graph(
+    def build_gnn(
         self,
-        node_column: str = None,
-        weight_column: str = None,
         X_nodes: XSymbolic = None,
         X_edges: XSymbolic = None,
         y_nodes: YSymbolic = None,
         y_edges: YSymbolic = None,
-        use_node_scaler: str = None,
-        use_edge_scaler: str = None,
+        weight_column: str = None,
+        train_split: float = 0.8,
```

Comment:
as we do more models, pretty sure we'll build in more modes here typical to graph problems